pkg/plugins/oauth/models.go

```diff
@@ -0,0 +1,37 @@
+package oauth
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+)
+
+// SelfCfg is a subset of oauthserver.SelfCfg making some fields optional
+type SelfCfg struct {
+	Enabled     *bool                      `json:"enabled,omitempty"`
+	Permissions []accesscontrol.Permission `json:"permissions,omitempty"`
+}
+
+// ImpersonationCfg is a subset of oauthserver.ImpersonationCfg making some fields optional
+type ImpersonationCfg struct {
+	Enabled     *bool                      `json:"enabled,omitempty"`
+	Groups      *bool                      `json:"groups,omitempty"`
+	Permissions []accesscontrol.Permission `json:"permissions,omitempty"`
+}
+
+// PluginExternalServiceRegistration is a subset of oauthserver.ExternalServiceRegistration
+// simplified for the plugin use case.
+type PluginExternalService struct {
+	Impersonation ImpersonationCfg `json:"impersonation"`
+	Self          SelfCfg          `json:"self"`
+}
+
+type PluginExternalServiceRegistration struct {
+	ClientID     string `json:"clientId"`
+	ClientSecret string `json:"clientSecret"`
+	PrivateKey   string `json:"privateKey"`
+}
+
+type ExternalServiceRegister interface {
+	SavePluginExternalService(ctx context.Context, name string, svc *PluginExternalService) (*PluginExternalServiceRegistration, error)
```

Comment:
Nit. Maybe rename `ExternalServiceRegister` to `ExternalServiceRegistry` and `SavePluginExternalService` to `RegisterExternalService` since this is already in the pkg/plugins/oauth package so "plugin" might be repetition of things? If changing would suggest renaming `PluginExternalService` struct as well.

Reply:
The oauth one is oauthserver.OAuth2Server.SaveExternalService and should not be confusing and import path enough I think: https://github.com/grafana/grafana/blob/813ee22b4de2f7fada635daa1be2ec20b045a7fa/pkg/services/oauthserver/models.go#L29-L32